legacy/firmware/transaction.c

```diff
@@ -684,19 +684,13 @@ uint32_t tx_serialize_middle_hash(TxStruct *tx) {
 uint32_t tx_serialize_footer(TxStruct *tx, uint8_t *out) {
   memcpy(out, &(tx->lock_time), 4);
 #if !BITCOIN_ONLY
-  if (tx->overwintered) {
-    if (tx->version == 3) {
-      memcpy(out + 4, &(tx->expiry), 4);
-      out[8] = 0x00;  // nJoinSplit
-      return 9;
-    } else if (tx->version == 4) {
-      memcpy(out + 4, &(tx->expiry), 4);
-      memzero(out + 8, 8);  // valueBalance
-      out[16] = 0x00;       // nShieldedSpend
-      out[17] = 0x00;       // nShieldedOutput
-      out[18] = 0x00;       // nJoinSplit
-      return 19;
-    }
+  if (tx->overwintered && tx->version == 4) {
```

Comment:
Can we rename the `overwintered` field to `is_zcashlike`?

I previously mistaken the struct for the protobuf one where field with this name is deprecated. However this is `TxStruct` which is used only in `transaction.c` and `signing.c`.

Reply:
I don't mind the change

Reply:
Renamed in 3f6370becf11dbcebb03c5a71d8aebcd7b952785.

I've also refactored the logic a little, let me know if it's out of scope & I'll drop it.